src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvokerCache.cs

```diff
@@ -172,29 +188,47 @@ public InnerCache(int version)
                 Version = version;
             }
 
-            public ConcurrentDictionary<ActionDescriptor, CacheEntry> Entries { get; } = 
-                new ConcurrentDictionary<ActionDescriptor, CacheEntry>();
+            public ConcurrentDictionary<ActionDescriptor, ControllerActionInvokerCacheEntry> Entries { get; } = 
+                new ConcurrentDictionary<ActionDescriptor, ControllerActionInvokerCacheEntry>();
 
             public int Version { get; }
         }
 
-        private struct CacheEntry
+        public struct ControllerActionInvokerCacheEntry
```

Comment:
If this is a nested type, just call it `Entry` - `ControllerActionInvokerCache` is already part of the name.